cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/AlertaSelfHealingNotifier.java

```diff
@@ -0,0 +1,250 @@
+                  alertaMessage.setCreateTime(outFormatter.format(Instant.ofEpochMilli(topicAnomaly.detectionTimeMs())));
+                  alertaMessages.add(alertaMessage);
+                });
+              }
+            } else {
+              AlertaMessage alertaMessage = new AlertaMessage(localHostname,
+                  anomalyType.toString());
+              alertaMessage.setSeverity(severityFromPriority(anomalyType.priority()));
+              alertaMessage.setGroup("Performance");
+              alertaMessage.setCreateTime(outFormatter.format(Instant.ofEpochMilli(topicAnomaly.detectionTimeMs())));
+              alertaMessages.add(alertaMessage);
+            }
+            break;
+          case MAINTENANCE_EVENT:
+            MaintenanceEvent maintenanceEvent = (MaintenanceEvent) anomaly;
+            AlertaMessage alertaMessage = new AlertaMessage(localHostname, anomalyType.toString() + " - " + maintenanceEvent.reasonSupplier());
+            alertaMessage.setSeverity(severityFromPriority(anomalyType.priority()));
+            alertaMessage.setGroup("Performance");
+            alertaMessage.setCreateTime(outFormatter.format(Instant.ofEpochMilli(maintenanceEvent.detectionTimeMs())));
+            alertaMessages.add(alertaMessage);
+            break;
+          default:
+            throw new IllegalStateException("Unrecognized anomaly type.");
+        }
+
+        for (AlertaMessage alertaMessage : alertaMessages) {
+          alertaMessage.setEnvironment(_alertaEnvironment);
+          alertaMessage.setService(Arrays.asList("cruise-control"));
+          alertaMessage.setText(text);
+          alertaMessage.setOrigin("cruise-control/" + localHostname);
+          alertaMessage.setType("cruiseControlAlarm");
+          alertaMessage.setRawData(anomaly.toString());
+          alertaMessage.setTags(Arrays.asList("alarm_id:" + anomaly.anomalyId()));
+          try {
+              sendAlertaMessage(alertaMessage);
+          } catch (IOException e) {
+              LOG.warn("ERROR sending alert to Alerta.io", e);
+          }
+        }
+    }
+
+    protected void sendAlertaMessage(AlertaMessage alertaMessage) throws IOException {
+        CloseableHttpClient client = HttpClients.createDefault();
+        HttpPost httpPost = new HttpPost(_alertaApiUrl + "/alert");
+        StringEntity entity = new StringEntity(alertaMessage.toString());
+        httpPost.setEntity(entity);
+        httpPost.setHeader("Authorization", "Key " + _alertaApiKey);
+        httpPost.setHeader("Accept", "application/json");
+        httpPost.setHeader("Content-type", "application/json");
+        try {
+            LOG.debug("Sending alert to Alerta.io : {}\nBody:\n{}", httpPost, alertaMessage);
+            CloseableHttpResponse httpResponse = client.execute(httpPost);
+            LOG.debug("Alerta.io response status: {}", httpResponse.getStatusLine().getStatusCode());
+        } finally {
+            client.close();
+        }
+    }
```

Comment:
Given that we have similar code in `SlackSelfHealingNotifier`, would it make sense to 
1. Create a utility class under `notifier` package,
2. Extract the common logic to the utility class?

Reply:
Done